pandas/core/dtypes/cast.py

```diff
@@ -1483,7 +1483,7 @@ def find_common_type(types: List[DtypeObj]) -> DtypeObj:
     if has_bools:
         for t in types:
             if is_integer_dtype(t) or is_float_dtype(t) or is_complex_dtype(t):
-                return np.object
+                return object
```

Comment:
```suggestion
                return np.dtype("object")
```

(to ensure this function actually returns a dtype object, as its type annotation indicates)